artiq/coredevice/comm_kernel.py

```diff
@@ -43,15 +43,116 @@ class Reply(Enum):
 class UnsupportedDevice(Exception):
     pass
 
+
 class LoadError(Exception):
     pass
 
+
 class RPCReturnValueError(ValueError):
     pass
 
 
 RPCKeyword = namedtuple('RPCKeyword', ['name', 'value'])
 
+receivers = {
```

Comment:
I suggest moving this at the bottom and merging it with the ``receivers[...]`` assignments. This way there is a single list with everything.

It is OK if ``receivers`` is referenced in the ``receive_...`` functions, this symbol resolution is done at runtime and by the time the functions are executed ``receivers`` exists in the namespace.